src/pds/peppi/query_builder.py

```diff
@@ -255,6 +288,41 @@ def bundles(self):
         self._add_clause(clause)
         return self
 
+    def contexts(self, keyword: str = None):
+        """Adds a query clause selecting only "Context" type products (targets, investigations, instruments, etc...).
+
+        Parameters
+        ----------
+        keyword : str, optional
+            Title of the context products. Automatically search for "cannonicalized" variations
+            (uppercase, lowercase, etc.) to cast a wider search across target names.
+
+        Returns
+        -------
+        This instance with the "Product Context" filter applied.
+
+        """
+        clause = 'product_class eq "Product_Context"'
+        self._add_clause(clause)
+
+        if keyword:
+
+            def _canonicalize_string(string: str):
+                return string.title(), string.upper(), string.lower()
+
+            def eq_cannonical_string_clause(property_name: str, value: str):
+                return " or ".join(f'{property_name} eq "{s}"' for s in _canonicalize_string(value))
+
+            q_string = eq_cannonical_string_clause("pds:Identification_Area.pds:title", keyword)
+            q_string += f"or {eq_cannonical_string_clause('pds:Alias.pds:alternate_title', keyword)}"
```

Comment:
Is a leading space before `or` needed here?

Reply:
I think that works without but that will be cleaner with.